improver/cube_combiner.py

```diff
@@ -52,6 +52,97 @@
+        new_name: str = None,
+        broadcast_to_threshold: bool = False,
+    ):
+        r"""
+        Args:
+            operation (str):
+                An operation to use in combining input cubes. One of:
+                +, -, \*, add, subtract, multiply, min, max, mean
+            new_name (str):
+                New name for the resulting dataset.
+            broadcast_to_threshold (bool):
+                If True, broadcast input cubes to the threshold coord prior to combining -
+                a threshold coord must already exist on the first input cube.
+            minimum_realizations (int):
+                If specified, the input cubes will be filtered to ensure that only realizations that
+                include all available lead times are combined. If the number of realizations that
+                meet this criteria are fewer than this integer, an error will be raised.
+                Minimum value is 1.
+        """
+        try:
+            self.minimum_realizations = int(minimum_realizations)
+        except TypeError:
+            if minimum_realizations is not None:
+                raise
+            self.minimum_realizations = None
+        self.new_name = new_name
+        self.broadcast_to_threshold = broadcast_to_threshold
+
+        if operation == "*" or operation == "multiply":
+            self.plugin = CubeMultiplier(
+                broadcast_to_threshold=self.broadcast_to_threshold
+            )
+        else:
+            self.plugin = CubeCombiner(operation)
+
+    def process(self, cubes: CubeList) -> Cube:
+        """
+        Preprocesses the cubes, then passes them to the appropriate plugin
+
+        Args:
+            cubes (iris.cube.CubeList or list of iris.cube.Cube):
+                An iris CubeList to be combined.
+
+        Returns:
+            result (iris.cube.Cube):
+                Returns a cube with the combined data.
+
+        Raises:
+            ValueError:
+                If minimum_realizations aren't met, or less than one were requested.
+        """
+        if not cubes:
+            raise TypeError("A cube is needed to be combined.")
+        if self.new_name is None:
+            self.new_name = cubes[0].name()
+
+        if self.minimum_realizations is not None:
+            if self.minimum_realizations < 1:
+                raise ValueError(
+                    f"Minimum realizations must be at least 1, not {self.minimum_realizations}"
+                )
+            from improver.utilities import FilterRealizations
+
+            cube = FilterRealizations()(cubes)
+            realization_count = len(cube.coord("realization").points)
+            if realization_count < self.minimum_realizations:
+                raise ValueError(
+                    f"After filtering, number of realizations {realization_count} "
+                    f"is less than {self.minimum_realizations}"
+                )
+            filtered_cubes = cube.slices_over("time")
```

Comment:
I can't decide if the implicit assumption here, that combine using filter realizations will only be used for accumulating diagnostics over time, is problematic or not. I can't come up with any use cases for it where we won't be doing that, though I had murky thoughts around whether it would be useful for mitigating against mismatches in the number of percentiles in phaseprob and preciprate, which are combined at a single time point. Any thoughts?

Reply:
Yes, this is the assumption. The doc-string describes the behaviour as applying to time coordinates, so users are on their own if they activate this argument with cubes that mismatch in some other way.